x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/AutoFollowCoordinator.java

```diff
@@ -47,22 +50,32 @@
     private final TimeValue pollInterval;
     private final ThreadPool threadPool;
     private final ClusterService clusterService;
+    private final CcrLicenseChecker ccrLicenseChecker;
 
     private volatile boolean localNodeMaster = false;
 
-    public AutoFollowCoordinator(Settings settings,
-                                 Client client,
-                                 ThreadPool threadPool,
-                                 ClusterService clusterService) {
+    public AutoFollowCoordinator(
+            Settings settings,
+            Client client,
+            ThreadPool threadPool,
+            ClusterService clusterService,
+            CcrLicenseChecker ccrLicenseChecker) {
         this.client = client;
         this.threadPool = threadPool;
         this.clusterService = clusterService;
+        this.ccrLicenseChecker = Objects.requireNonNull(ccrLicenseChecker, "ccrLicenseChecker");
 
         this.pollInterval = CcrSettings.CCR_AUTO_FOLLOW_POLL_INTERVAL.get(settings);
         clusterService.addStateApplier(this);
     }
 
     private void doAutoFollow() {
+        if (ccrLicenseChecker.isCcrAllowed() == false) {
+            // TODO: set non-compliant status on auto-follow coordination that can be viewed via a stats API
```

Comment:
:+1: Especially because this cluster could be following multiple leader clusters.